### PR TITLE
Add support for 'using' statements that bring in a single definition

### LINKED ISSUE #19: Doesn't resolve namespaces for types imported via `using`
```
#include <string>
namespace a {
    using std::string;
    using VoidFunction = std::function<void()>;

    void fn(string &s, VoidFunction fn);
}
```
It needs to tack on the namespace to the string/VoidFunction types. This does not cover `using namespace foo`, as that isn't doable using the information available to CppHeaderParser.

### PR DESCRIPTION
**Pull request by virtuald**
- Fixes #19

### CODE REVIEW

File: CppHeaderParser/CppHeaderParser.py [hunk 1]
```diff
@@ -377,6 +377,37 @@ def filter_out_attribute_keyword(stack):
         return stack
 
 
+_nhack = re.compile(r"[A-Za-z_][A-Za-z0-9_]*")
```
Comment (auscompgeek) [inline CppHeaderParser/CppHeaderParser.py]: Could probably just use `\w` for the second class here.

File: CppHeaderParser/CppHeaderParser.py [hunk 2]
```diff
@@ -377,6 +377,37 @@ def filter_out_attribute_keyword(stack):
...
+            "".join(namestack[: last_colon + 1]),
+            namestack[last_colon + 1 :],
+        )
+    else:
+        ns = ""
+
+    return ns, namestack
+
+
+def _iter_ns_str_reversed(namespace):
```
Comment (auscompgeek) [inline CppHeaderParser/CppHeaderParser.py]: This isn't terribly obvious from the name what this does. It took me about 5 minutes to work out what this is for. Mind adding a docstring?

File: CppHeaderParser/CppHeaderParser.py [hunk 3]
```diff
@@ -377,6 +377,37 @@ def filter_out_attribute_keyword(stack):
         return stack
 
 
+_nhack = re.compile(r"[A-Za-z_][A-Za-z0-9_]*")
+
+
+def _extract_namespace(namestack):
```
Comment (auscompgeek) [inline CppHeaderParser/CppHeaderParser.py]: Maybe `_split_namespace` might be a better name?

File: CppHeaderParser/CppHeaderParser.py (hunk 1 above)
Comment (virtuald) [reply CppHeaderParser/CppHeaderParser.py]: The other regexes in here aren't \w, so I elected to copy those. I think I'd rather fix them all at the same time, some other time.

**Pull request merged**